homeassistant/components/homematic/light.py

```diff
@@ -62,30 +60,48 @@ def is_on(self):
             return False
 
     @property
-    def supported_features(self):
-        """Flag supported features."""
-        features = SUPPORT_BRIGHTNESS | SUPPORT_TRANSITION
+    def color_mode(self) -> str:
+        """Return the color mode of the light."""
+        if "COLOR" in self._hmdevice.WRITENODE:
+            return COLOR_MODE_HS
+        if hasattr(self._hmdevice, "get_color_temp"):
+            return COLOR_MODE_COLOR_TEMP
+        return COLOR_MODE_BRIGHTNESS
+
+    @property
+    def supported_color_modes(self) -> set[str] | None:
+        """Flag supported color modes."""
+        color_modes = set()
 
         if "COLOR" in self._hmdevice.WRITENODE:
-            features |= SUPPORT_COLOR
+            color_modes.add(COLOR_MODE_HS)
+        if hasattr(self._hmdevice, "get_color_temp"):
+            color_modes.add(COLOR_MODE_COLOR_TEMP)
+        if not color_modes:
+            color_modes.add(COLOR_MODE_BRIGHTNESS)
+
+        return color_modes
```

Comment:
This code is assuming as if:
- A homematic light's features can dynamically change, and is not necessarily known when the `HMLight` object instantiated
- There are homematic lights which can support both hs color and color temperature

Looking at the code in pyhomematic, I think neither of this is true; `IPKeySwitchLevel` and `ColorEffectLight` supports hs color, but not color temperature, `ColdWarmDimmer` supports color temperature but not hs color and there's no light supporting both.

Reply:
I don't know about the second one but the first one is wrong. Nothing is dynamic, all is known after the `HMLight` object is created